rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java

```diff
@@ -77,9 +88,14 @@ public class DataSourceProxy extends AbstractDataSourceProxy implements Resource
     private static final long TABLE_META_CHECKER_INTERVAL = ConfigurationFactory.getInstance().getLong(
             ConfigurationKeys.CLIENT_TABLE_META_CHECKER_INTERVAL, DEFAULT_TABLE_META_CHECKER_INTERVAL);
 
-    private final ScheduledExecutorService tableMetaExecutor = new ScheduledThreadPoolExecutor(1,
+    private final ScheduledExecutorService tableMetaCheckExecutor = new ScheduledThreadPoolExecutor(1,
         new NamedThreadFactory("tableMetaChecker", 1, true));
 
+    private final Executor tableMetaRefreshExecutor = new ThreadPoolExecutor(1, 1,
```

Comment:
Both the timed refresh and the exception refresh themselves are refreshes to tableMeta, so it is necessary to separate two threads?

Reply:
Is DataSourceProxy reasonably responsible for refreshing and should be moved to TableMetaCache?